Assets/Scripts/Enemy.cs

```diff
@@ -46,12 +52,31 @@ private void OnTriggerEnter(Collider other)
     {
         if (other.gameObject.tag == "Player-Fireball")
         {
-            ChangeHealth(-1);
-            Destroy(other.gameObject);
+            if (Health != 0)
+            {
+                ChangeHealth(-1);
+                Destroy(other.gameObject);
         }
+    }
 
     }
 
+    void HealthDrop()
+    {
+        // calculate whether or not to drop
+        var chance = Random.Range(0f, 1f);
+        if(chance <= healthDropProbability)
+        {
+            var randX = Random.Range(-600, 600);
```

Comment:
Question: could this potentially throw a heart across the room?

Reply:
No, the values shouldn't be high enough.